lib/image.php

```diff
@@ -260,6 +260,26 @@ function __errorHandler($errno=NULL, $errstr, $errfile=NULL, $errline=NULL, $err
 		$etag = NULL;
 	}
 
+	// Allow CORS
+	// respond to preflights
+	if ($settings['image']['allow_origin'] !== null) {
+		if ($_SERVER['REQUEST_METHOD'] == 'OPTIONS') {
+			// return only the headers and not the content
+			// only allow CORS if we're doing a GET - i.e. no sending for now.
+			if (isset($_SERVER['HTTP_ACCESS_CONTROL_REQUEST_METHOD']) && $_SERVER['HTTP_ACCESS_CONTROL_REQUEST_METHOD'] == 'GET') {
+				header('Access-Control-Allow-Origin: *');
+				header('Access-Control-Allow-Headers: X-Requested-With');
+			}
+			exit;
+		} else {
+			header('Origin: ' . $settings['image']['allow_origin']);
+			header('Access-Control-Allow-Origin: ' . $settings['image']['allow_origin']);
+			header('Access-Control-Allow-Methods: GET');
+			header('Access-Control-Max-Age: 3000');
+			header('Vary: Origin, Access-Control-Request-Headers, Access-Control-Request-Method');
```

Comment:
Don't remember anymore if there was a good reason… sry.

> On 21 Mar 2016, at 23:05, Nicolas Brassard <notifications@github.com> wrote:
> 
> In lib/image.php:
> 
> > +	// respond to preflights
> > +	if ($settings['image']['allow_origin'] !== null) {
> > +		if ($_SERVER['REQUEST_METHOD'] == 'OPTIONS') {
> > +			// return only the headers and not the content
> > +			// only allow CORS if we're doing a GET - i.e. no sending for now.
> > +			if (isset($_SERVER['HTTP_ACCESS_CONTROL_REQUEST_METHOD']) && $_SERVER['HTTP_ACCESS_CONTROL_REQUEST_METHOD'] == 'GET') {
> > +				header('Access-Control-Allow-Origin: *');
> > +				header('Access-Control-Allow-Headers: X-Requested-With');
> > +			}
> > +			exit;
> > +		} else {
> > +			header('Origin: ' . $settings['image']['allow_origin']);
> > +			header('Access-Control-Allow-Origin: ' . $settings['image']['allow_origin']);
> > +			header('Access-Control-Allow-Methods: GET');
> > +			header('Access-Control-Max-Age: 3000');
> > +			header('Vary: Origin, Access-Control-Request-Headers, Access-Control-Request-Method');
> Vary header is buggy I would remove it. @tmslnz a particular reason on why you added it ?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>